tough/src/schema/decoded.rs

```diff
@@ -126,6 +126,36 @@ impl Encode for EcdsaPem {
     }
 }
 
+/// [`Decode`]/[`Encode`] implementation for ECDSA public keys.
+/// This is a flexible implementation, it will try to decode the key assuming
+/// it is PEM encoded, if the decode fails it will then try to decode it
+/// assuming it's Hex encoded.
+/// The official TUF specification suggests ECDSA keys to be PEM encoded,
+/// however the go-tuf implementation encodes them as Hex numbers.
+/// This flexible decoder tries to cover both cases in a transparent way.
+#[derive(Debug, Clone, Copy)]
+pub struct EcdsaFlex {}
+
+impl Decode for EcdsaFlex {
+    fn decode(s: &str) -> Result<Vec<u8>, Error> {
+        if s.starts_with("-----BEGIN ") {
+            spki::decode(
+                spki::OID_EC_PUBLIC_KEY,
+                Some(spki::OID_EC_PARAM_SECP256R1),
+                s,
+            )
```

Comment:
Can this re-use EcdsaPem::decode?
```suggestion
            EcdsaPem::decode(s)
```